src/main/java/org/folio/clients/SynchronousOkapiClient.java

```diff
@@ -0,0 +1,63 @@
+package org.folio.clients;
+
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TOKEN;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.ws.rs.core.MediaType;
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.folio.util.OkapiConnectionParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Base class for okapi clients that use blocking method for http calls.
+ */
+public abstract class SynchronousOkapiClient {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  protected final OkapiConnectionParams okapiConnectionParams;
+
+  public SynchronousOkapiClient(OkapiConnectionParams okapiConnectionParams) {
+    this.okapiConnectionParams = okapiConnectionParams;
+  }
+
+  public Optional<JsonObject> getById(String id) {
+    HttpGet httpGet = new HttpGet();
```

Comment:
here we are establishing a new HTTP Connection each time we make a get request. I think we can use one connection, and close it after all the calls are made

Reply:
Changed.